test/syntax.jl

```diff
@@ -1968,7 +1968,7 @@ let a(; b) = b
 end
 
 # issue #33987
-f33987(args::(Vararg{Any, N} where N); kwargs...) = args
+f33987(args::Vararg{Any, N}; kwargs...) where N = args
```

Comment:
This doesn't test the same thing. Might as well delete the test, although it should keep working as long as the deprecation is in place.

Reply:
Fair. I'd changed this before I put in the deprecation, so I'll change it back to still get the parsing case.